v0/destinations/singular/transform.js

```diff
@@ -0,0 +1,159 @@
+const get = require("get-value");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  BASE_URL,
+  sessionEvents,
+  SINGULAR_SESSION_EXCLUSION,
+  SINGULAR_EVENT_EXCLUSION
+} = require("./config");
+const {
+  defaultRequestConfig,
+  constructPayload,
+  CustomError,
+  defaultGetRequestConfig,
+  removeUndefinedAndNullAndEmptyValues,
+  isAppleFamily,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  extractCustomFields
+} = require("../../util");
+
+const responseBuilderSimple = (message, { Config }) => {
+  let eventType = message.event;
+  // trim and replace spaces with "_"
+  eventType = eventType.trim().replace(/\s+/g, " ");
+  const response = defaultRequestConfig();
+
+  /* eslint no-param-reassign: "error" */
+  const setPayload = payload => {
+    if (isAppleFamily(payload.p)) {
+      payload.idfa = get(message, "context.device.advertisingId");
+      payload.idfv = get(message, "context.device.id");
+      /*
+        Android - Captured from SDK. No need to pass it explicitly
+        iOS - No need to capture from SDK. Need to pass it explicitly under properties.ua.
+       */
+      payload.ua = get(message, "properties.ua");
+    } else if (payload.p.toLowerCase() === "android") {
+      payload.aifa = get(message, "context.device.advertisingId");
+      payload.andi = get(message, "context.device.id");
+    }
+    return payload;
+  };
+
+  if (sessionEvents.includes(eventType)) {
+    let sessionPayload = constructPayload(
+      message,
+      MAPPING_CONFIG[CONFIG_CATEGORIES.SESSION.name]
+    );
+    if (!sessionPayload) {
+      // fail-safety for developer error
+      throw new CustomError("Failed to Create Session Payload", 400);
+    }
```

Comment:
Do we need this check at all as some of the attributes are required in the config.json files?